crates/turborepo-cache/src/multiplexer.rs

```diff
@@ -73,9 +75,14 @@ impl CacheMultiplexer {
 
         let http_result = match self.get_http_cache() {
             Some(http) => {
-                let http_result = http.put(anchor, key, files, duration).await;
+                if self.read_only_remote_cache {
+                    let http_result = http.put(anchor, key, files, duration).await;
 
-                Some(http_result)
+                    Some(http_result)
+                }
+                else {
+                    None
```

Comment:
@mehulkar Should we print a message for each task that runs into this? Or just once for the entire invocation? If you have an examples of where we do one or the other let me know

Reply:
Once per invocation seems good to me. I guess you could print the message up front instead of on throw from here

Still seems better to throw and catch upstream, but will defer to Rustaceans @chris-olszewski and especially @NicholasLYang who's working on error flows in case they have a better idea of a good pattern.

Reply:
@NicholasLYang might have differing opinions, but I think a `warn!` message along the lines of "not writing to remote cache" makes sense here since the operation was partially successful. (Unless someone is using `--remote-only` and `--remote-cache-read-only` and then nothing would be written to any cache).

Largest thing we want is to avoid printing the warning message for ever task that's run. You can probably get away with adding a `should_print_skipping_remote_put: AtomicBool` to the multiplexer and only printing it if it's true and then updating it to false once it has been printed once.

Reply:
ok cool so similar to the `self.should_use_http_cache.store` approach right below it 👍🏼 what about the throw then catch bit? do I need to throw an exception and catch it or can I just set the bool right in that block?

Reply:
https://github.com/vercel/turbo/pull/6326/commits/b169c55ad7a2a3b1973d07c6b3250409b45a88d4

@mehulkar @chris-olszewski think I've implemented that more or less as-described. Is there an easy way for me to run the turbo binary locally with these changes? Or does CI output an Artifact I could download and test with?

Reply:
> do I need to throw an exception and catch it or can I just set the bool right in that block

You can just set it right in the block.

> Is there an easy way for me to run the turbo binary locally with these changes?

You can produce a binary with `cargo build -p turbo` from the repository and then test out the changes using by using the generated binary located `target/debug/turbo`. You can use it with the `--skip-infer` flag to avoid using the local `turbo` binary that's located in your repository. e.g. `~/code/turbo/target/debug/turbo --skip-infer build --remote-cache-read-only`